src/storage/txn/sched_pool.rs

```diff
@@ -102,94 +183,103 @@ impl SchedPool {
                     tls_flush(&reporter);
                 })
         };
-        // FIXME: for performance issue, disable priority pool temporarily
-        // if let Some(ref r) = resource_ctl {
-        //     SchedPool::Priority {
-        //         worker_pool: builder(pool_size, "sched-worker-pool")
-        //             .build_priority_future_pool(r.clone()),
-        //         resource_ctl: r.clone(),
-        //     }
-        // } else {
-        SchedPool::Vanilla {
+        let vanilla = VanillaQueue {
             worker_pool: builder(pool_size, "sched-worker-pool").build_future_pool(),
-            high_worker_pool: builder(std::cmp::max(1, pool_size / 2), "sched-high-pri-pool")
+            high_worker_pool: builder(std::cmp::max(1, pool_size / 2), "sched-worker-high")
                 .build_future_pool(),
+        };
+        let priority = if let Some(ref r) = resource_ctl {
+            Some(PriorityQueue {
+                worker_pool: builder(pool_size, "sched-worker-priority")
+                    .build_priority_future_pool(r.clone()),
+                resource_ctl: r.clone(),
+            })
+        } else {
+            None
+        };
+        let queue_type = if resource_ctl.is_some() {
+            info!("sched-worker pool use dynamic mode");
+            QueueType::Dynamic
+        } else {
+            info!("sched-worker pool use vanilla mode");
+            QueueType::Vanilla
+        };
+
+        SchedPool {
+            vanilla: Some(vanilla),
+            priority,
+            queue_type,
         }
-        // }
     }
 
     pub fn spawn(
         &self,
         group_name: &str,
-        priority: CommandPri,
+        priority_level: CommandPri,
         f: impl futures::Future<Output = ()> + Send + 'static,
     ) -> Result<(), Full> {
-        match self {
-            SchedPool::Vanilla {
-                high_worker_pool,
-                worker_pool,
-            } => {
-                if priority == CommandPri::High {
-                    high_worker_pool.spawn(f)
+        match self.queue_type {
+            QueueType::Vanilla => {
+                let vanilla = self.vanilla.as_ref().unwrap();
+                vanilla.spawn(priority_level, f)
+            }
+            QueueType::Priority => {
```

Comment:
Seems queue type never be QueueType::Priority?

Reply:
yes, user only uses `Dynamic` and `Vanilla`. but for completeness(some places need it), I remain it.

Reply:
exactly no place to use it, but the inner use PriorityQueue.

Reply:
removed.